rustls/src/crypto/ring/quic.rs

```diff
@@ -0,0 +1,401 @@
+use crate::crypto::cipher::Iv;
+use crate::error::Error;
+use crate::hkdf;
+use crate::quic;
+use crate::tls13::key_schedule::{hkdf_expand, hkdf_expand_aead_key};
+use crate::tls13::Tls13CipherSuite;
+
+use ring::aead;
+
+pub(crate) struct HeaderProtectionKey(aead::quic::HeaderProtectionKey);
+
+impl HeaderProtectionKey {
+    pub(crate) fn new(
+        expander: &hkdf::Expander,
+        version: quic::Version,
+        alg: &'static aead::quic::Algorithm,
+    ) -> Self {
+        let key = hkdf_expand_aead_key(expander, alg.key_len(), version.header_key_label(), &[]);
+        Self(aead::quic::HeaderProtectionKey::new(alg, key.as_ref()).unwrap())
+    }
+
+    fn xor_in_place(
+        &self,
+        sample: &[u8],
+        first: &mut u8,
+        packet_number: &mut [u8],
+        masked: bool,
+    ) -> Result<(), Error> {
+        // This implements [Header Protection Application] almost verbatim.
```

Comment:
I think this might be a dead docs link now. The definition was in `rustls/src/quic.rs` but didn't make the move to `ring/quic.rs`

Reply:
have fixed this, though note it's not a comment that would be processed by `cargo doc` anyway.